libfs/file_info.go

```diff
@@ -82,39 +88,62 @@ type fileInfoSys struct {
 	fi *FileInfo
 }
 
-var _ LastWriterGetter = fileInfoSys{}
+type KBFSMetadataForSimpleFS struct {
+	LastWriter     keybase1.User
+	PrefetchStatus keybase1.PrefetchStatus
+}
+
+var _ KBFSMetadataForSimpleFSGetter = fileInfoSys{}
 
-func (fis fileInfoSys) LastWriter() (keybase1.User, error) {
+func (fis fileInfoSys) KBFSMetadataForSimpleFS() (KBFSMetadataForSimpleFS,
+	error) {
 	if fis.fi.node == nil {
 		// This won't return any last writer for symlinks themselves.
 		// TODO: if we want symlink last writers, we'll need to add a
 		// new interface to KBFSOps to get them.
-		return keybase1.User{}, nil
+		return KBFSMetadataForSimpleFS{}, nil
 	}
 	md, err := fis.fi.fs.config.KBFSOps().GetNodeMetadata(
 		fis.fi.fs.ctx, fis.fi.node)
 	if err != nil {
-		return keybase1.User{}, err
+		return KBFSMetadataForSimpleFS{}, err
 	}
+	var prefetchStatus keybase1.PrefetchStatus
+	switch md.PrefetchStatus {
+	case "NoPrefetch":
+		prefetchStatus = keybase1.PrefetchStatus_NOT_STARTED
+	case "TriggeredPrefetch":
+		prefetchStatus = keybase1.PrefetchStatus_IN_PROGRESS
+	case "FinishedPrefetch":
+		prefetchStatus = keybase1.PrefetchStatus_COMPLETE
+	case "Unknown":
+		return KBFSMetadataForSimpleFS{}, ErrUnknownPrefetchStatus
+	default:
+		return KBFSMetadataForSimpleFS{}, ErrUnknownPrefetchStatus
+	}
+
 	lastWriterName := md.LastWriterUnverified
 	if lastWriterName == "" {
 		// This can happen in old, buggy team folders where the writer
 		// isn't properly set.  See KBFS-2939.
-		return keybase1.User{}, nil
+		return KBFSMetadataForSimpleFS{PrefetchStatus: prefetchStatus}, nil
 	}
 
 	_, id, err := fis.fi.fs.config.KBPKI().Resolve(
 		fis.fi.fs.ctx, lastWriterName.String())
 	if err != nil {
-		return keybase1.User{}, err
+		return KBFSMetadataForSimpleFS{PrefetchStatus: prefetchStatus}, nil
```

Comment:
I think this and the next `return` should return `KBFSMetadataForSimpleFS{}, err`. Otherwise, the error is lost.  And I think it's ok to lose the prefetch status in those rare cases.